Description: Transitional patch from jna-posix to jnr-posix as
 currently packaged in Debian and Ubuntu.
Author: James Page <james.page@ubuntu.com>
Forwarded: not-needed

Index: jenkins/core/src/main/java/hudson/os/PosixAPI.java
===================================================================
--- jenkins.orig/core/src/main/java/hudson/os/PosixAPI.java	2012-07-24 16:32:51.226005626 +0100
+++ jenkins/core/src/main/java/hudson/os/PosixAPI.java	2012-07-24 16:33:08.182066097 +0100
@@ -1,10 +1,9 @@
 package hudson.os;
 
-import org.jruby.ext.posix.JavaPOSIX;
 import org.jruby.ext.posix.POSIX;
 import org.jruby.ext.posix.POSIXFactory;
 import org.jruby.ext.posix.POSIXHandler;
-import org.jruby.ext.posix.POSIX.ERRORS;
+import com.kenai.constantine.platform.Errno;
 
 import java.io.File;
 import java.io.InputStream;
@@ -34,11 +33,11 @@
      * used a fallback java implementation which does not support many operations.
      */
     public static boolean supportsNative() {
-        return !(posix instanceof JavaPOSIX);
+        return posix.isNative();
     }
     
     private static final POSIX posix = POSIXFactory.getPOSIX(new POSIXHandler() {
-        public void error(ERRORS errors, String s) {
+        public void error(Errno errors, String s) {
             throw new PosixException(s,errors);
         }
 
Index: jenkins/core/src/main/java/hudson/os/PosixException.java
===================================================================
--- jenkins.orig/core/src/main/java/hudson/os/PosixException.java	2012-07-24 16:32:51.226005626 +0100
+++ jenkins/core/src/main/java/hudson/os/PosixException.java	2012-07-24 16:33:08.182066097 +0100
@@ -1,6 +1,6 @@
 package hudson.os;
 
-import org.jruby.ext.posix.POSIX.ERRORS;
+import com.kenai.constantine.platform.Errno;
 
 /**
  * Indicates an error during POSIX API call.
@@ -8,14 +8,14 @@
  * @author Kohsuke Kawaguchi
  */
 public class PosixException extends RuntimeException {
-    private final ERRORS errors;
+    private final Errno errors;
 
-    public PosixException(String message, ERRORS errors) {
+    public PosixException(String message, Errno errors) {
         super(message);
         this.errors = errors;
     }
 
-    public ERRORS getErrorCode() {
+    public Errno getErrorCode() {
         return errors;
     }
 
Index: jenkins/core/pom.xml
===================================================================
--- jenkins.orig/core/pom.xml	2012-07-24 16:33:07.086062188 +0100
+++ jenkins/core/pom.xml	2012-07-24 16:33:08.182066097 +0100
@@ -117,6 +117,11 @@
       <version>1.0.3</version>
     </dependency>
     <dependency>
+      <groupId>org.jruby.extras</groupId>
+      <artifactId>jffi</artifactId>
+      <version>0.6.5</version>
+    </dependency>
+    <dependency>
       <groupId>org.kohsuke</groupId>
       <artifactId>trilead-putty-extension</artifactId>
       <version>1.0</version>
